src/config.c

```diff
@@ -72,7 +72,7 @@ static int cfg_merge_local_section() {
 	if (!cfg_database) return 0;
 	if (strncmp(cfg_database->name, "local:", 6)) return 0;
 	if (!strncmp(cfg_database->name, "local:list.", 11)) return 0;
-	parent = cfg_get_section(&cfg_database->name[6], NULL);
+	parent = (struct cfg_section*)cfg_get_section(&cfg_database->name[6], NULL);
```

Comment:
Having such casts is kind of defeats the purpose of const'ification. Maybe we shouldn't be const'infying the `struct cfg_section *` pointers for now, and focus on `char *` ones only?

Reply:
Possibly. And we might want to consider reverting `path_expand_ex()` as well, if we keep it at all.